lib/linter.js

```diff
@@ -50,6 +49,40 @@ const MAX_AUTOFIX_PASSES = 10;
 //------------------------------------------------------------------------------
 
 /**
+ * Creates an object which can listen for and emit events.
```

Comment:
Can this be moved out into it's own file? Since the behavior doesn't have direct connection to Linter, I'd rather see a separate file, even if it's small.